src/flamenco/runtime/fd_executor.c

```diff
@@ -1100,6 +1105,402 @@ fd_execute_txn_finalize( fd_exec_slot_ctx_t * slot_ctx,
   return 0;
 }
 
+static void
+create_txn_context_protobuf_from_instructions( fd_exec_test_txn_context_t * txn_context_msg,
+                                               fd_exec_txn_ctx_t *          txn_ctx ) {
+  /*
+  NOTE: Caller of this function should have a scratch frame ready
+  */
+  fd_txn_t const * txn_descriptor = txn_ctx->txn_descriptor;
```

Comment:
ideally should go in the function comment which should document expected failure modes